docs/overview.rst

```diff
@@ -1,20 +1,110 @@
 Why new programming language
 ============================
 
-Taichi is a high-performance programming language for computer graphics applications. The design goals are
+Taichi is a high-performance programming language for computer graphics applications.
+
+Introduction
+------------
+
+Different backends
+******************
+
+Before Taichi has been invented, to write a **general purposed GPU (GPGPU)**
+program, we have to manipulate through tons of different APIs, e.g.:
+
+``cuDeviceSynchronize()`` for CUDA, ``glDispatchCompute()`` for OpenGL,
+``pthread_join()`` for multi-core CPUs...
+
+Downside:
+
+1. CUDA, for example, despite its powerfulness, can run only on supported NVIDIA
+   GPUs. One could not share/distribute his/her work with/to a computer
+   without NVIDIA GPUs.
+   This means the loss of portability.
```

Comment:
```suggestion
   without NVIDIA GPUs. This means the loss of portability.
```

Reply:
Maybe we don't need a line break here to keep it consistent with the following sections